sdk/test/utxo.test.ts

```diff
@@ -392,4 +415,57 @@ describe('UTXO Tests', () => {
             'Failed to create transaction. Do you have enough funds?'
         );
     });
+
+    it('should return address balance', async () => {
+        const address = 'bc1peqr5a5kfufvsl66444jm9y8qq0s87ph0zv4lfkcs7h40ew02uvsqkhjav0';
+
+        const balance = await getBalance(address);
+
+        assert(balance.confirmed);
+        assert(balance.total);
+        assert(
+            balance.confirmed === balance.total
+                ? balance.unconfirmed === 0n
+                : balance.unconfirmed === balance.total - balance.confirmed
+        );
+
+        const zeroBalance = await getBalance();
+
+        assert(zeroBalance.confirmed === 0n, 'If no address specified confirmed must be 0');
+        assert(zeroBalance.unconfirmed === 0n, 'If no address specified unconfirmed must be 0');
+        assert(zeroBalance.total === 0n, 'If no address specified total must be 0');
+    });
+
+    it('returns smalled amount if address holds ordinals', async () => {
+        const taprootAddress = 'bc1peqr5a5kfufvsl66444jm9y8qq0s87ph0zv4lfkcs7h40ew02uvsqkhjav0';
+
+        const esploraClient = new EsploraClient('mainnet');
+
+        const outputs = await esploraClient.getAddressUtxos(taprootAddress);
+
+        const total = outputs.reduce((acc, output) => acc + output.value, 0);
+
+        const confirmed = outputs.reduce((acc, output) => {
+            if (output.confirmed) {
+                return acc + output.value;
+            }
+
+            return acc;
+        }, 0);
+
+        // mock half of the UTXOs contain inscriptions or runes
+        (OrdinalsClient.prototype.getOutputsFromAddress as Mock).mockResolvedValueOnce(
+            outputs.slice(Math.ceil(outputs.length / 2)).map((output) => {
+                const outpoint = OutPoint.toString(output);
+
+                return { outpoint };
+            })
+        );
+
+        const balanceData = await getBalance(taprootAddress);
+
+        expect(balanceData.total).toBeLessThanOrEqual(total);
+        expect(balanceData.confirmed).toBeLessThanOrEqual(confirmed);
```

Comment:
Should it not be:
```suggestion
        expect(balanceData.confirmed).toBeLessThanOrEqual(confirmed/2);
```
etc.

Reply:
it's not that we are taking half of the confirmed amount but half of the confirmed UTXOs. theoretically total amount of that half could be more than half of the confirmed amount.

example (4 UTXOs and their values):
`1 1 3 4`
then if we take right half of that list`3 + 4 -> total 7` will be greater than `1 + 1 + 3 + 4) / 2 -> total 4.5`

Reply:
Ah right yes you are correct, but what I mean is that validating `toBeLessThanOrEqual` doesn't really tell us much right? Let's say for some reason the code we are unit testing is broken and allows spending all UTXOs including those containing runes or ordinals, then the test will still pass.

Reply:
notes for that test:
- fetching all confirmed UTXOs from explora client -- some of the might contain ordinals
- mocking response of ord api so that left half of that list contains ordinals (api will return only right path) -- (example above is incorrect, should be `4 3 1 1`, `4 3` - contain inscriptions, `1 1` - do not contain inscriptions)
- executing `getBalance` fn.

so if we are assuming that some of these UTXOs contain inscription then total value of `all` UTXOs should be less than or equal (if there're 0 UTXOs) to the total amount of UTXOs (cardinal) that ord api will respond with.

> Would be good to have a test that checks the logic i.e. returns a smaller number if the address then holds ordinals. Maybe just alter the mocks to not return one of the UTXOs or something?

Reply:
> Ah right yes you are correct, but what I mean is that validating `toBeLessThanOrEqual` doesn't really tell us much right? Let's say for some reason the code we are unit testing is broken and allows spending all UTXOs including those containing runes or ordinals, then the test will still pass.

So let's say ord api with `cardinal` query params returns the exact same UTXOs as not containing inscription (which is incorrect) then yes -- this behaviour is incorrect.

This test is not testing ord implementation of `/outputs/:address?type=cardinal` -- we are assuming it works properly all the time. Such behaviour ideally should be caught by ord unit tests.